### PR TITLE
Remove identity argument from UserProfile.signTransaction

### LINKED ISSUE #641: Remove identity arguments from UserProfile.signTransaction
The identity we pass must match the transaction creator. Thus the data is redundant and the `identity: PublicIdentity` argument can be removed.

Current interface

```ts
  public async signTransaction(
    id: WalletId,
    identity: PublicIdentity,
    transaction: UnsignedTransaction,
    codec: TxCodec,
    nonce: Nonce,
  ): Promise<SignedTransaction>
```

### PR DESCRIPTION
**Pull request by webmaster128**
Based on #806

Closes #641

### CODE REVIEW

**Comment by ethanfrey** [comment]
I was excited to see the walletId removed, then realized it was the identity.
Yes, this is a duplicate of info already in the transaction, so good change.

Waiting for walletId to disappear from the arg list too :crossed_fingers:

**Comment by webmaster128** [comment]
Rebased and ready for reaview

File: packages/iov-keycontrol/src/userprofile.ts [hunk 1]
```diff
@@ -192,27 +191,23 @@ export class UserProfile {
...
   public async signTransaction(
     id: WalletId,
-    identity: PublicIdentity,
     transaction: UnsignedTransaction,
     codec: TxCodec,
     nonce: Nonce,
   ): Promise<SignedTransaction> {
-    // identity and the pair (transaction.chainId, transaction.signer) are redundant
-    // but we keep both in the interface to be consistent with appendSignature() where
-    // the original transaction creator is not the signer
```
Comment (willclarktech) [inline packages/iov-keycontrol/src/userprofile.ts]: Why exactly is this no longer valid?

Comment (webmaster128) [reply packages/iov-keycontrol/src/userprofile.ts]: We decided that it is not important to keep the signature of signTransaction() and appendSignature() the same. In the simple cases (signTransaction), the transaction creator is always the one who signs the transaction. This saves us an additional, redundant argument.

**Pull request merged**